src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -343,6 +345,8 @@ private async Task ReadResponseAsync(CancellationToken cancellationToken)
             while ((int)_response.StatusCode < 200);
 
             _headerState = HeaderState.TrailingHeaders;
+
+            if (HttpTelemetry.Log.IsEnabled()) HttpTelemetry.Log.ResponseHeadersStop();
```

Comment:
This is not exactly the moment when the headers get sent, but it gets muddy with the gathered buffer with content. How does this compare with H/2 and H/1.1? Is is worth trying to refine the moment or not?

Reply:
On HTTP/2, we will actually include the time to write out the headers on the wire ([code](https://github.com/dotnet/runtime/blob/dc9979136cf0ff1addbc1a4c0f06ea4b65f002d4/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs#L1529-L1587)). That is because the time when the headers are sent is well-defined since we immediately serialize the action on the multiplexed connection.

On HTTP/1.1 and 3, however, the headers may not end up being sent right away, but only with the first flush of the request content, which may be under user's control.
We could report a more accurate timestamp of when the headers are actually sent, but these events being Start/Stop complicates the matter. If ThreadPool activity tracking is on, I don't see how we would get the async scope for Start/Stop to match, given Stop would result as part of the content copying outside of `SendAsync`.

Reply:
I guess this is good enough and not worth the trouble. Especially, if we're already not pedantic about the exact times in H/1.1